src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,10 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
It is not obvious why it is the right thing to do to filter on the arity here and how this is going to "ensure we attempt to perform type inference and bind a lambda expression argument". Could you please provide more details on this.
Also, why should the filtering be done when type arguments were provided explicitly, or successfully inferred?